pkg/action/output.go

```diff
@@ -87,3 +72,54 @@ func ParseOutputFormat(s string) (out OutputFormat, err error) {
 	}
 	return
 }
+
+// Writer is an interface that any type can implement to write supported formats
+type Writer interface {
```

Comment:
2 cents:  you could have the abstraction be `Write()` and types for table,json,yaml.  This would protect from having to add a new method to this interface to add a formatter.

Reply:
My thinking with this interface is that it will cause a compilation error if someone adds a new type and doesn't implement it for everything, but if you think this would be cleaner, I can give it a whirl

Reply:
Going to merge for now. If you think this would help, let me know and I can do a follow up PR